### PR TITLE
Move printer demo and energy-monitor into new top-level demos/ folder

### PR DESCRIPTION
**Pull request by tronical**
These are showing off use-cases for Slint, but they're not examples showing individual Slint features.

Also removed the old printerdemo while at it.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -21,15 +21,14 @@ members = [
...
     'examples/slide_puzzle',
     'examples/todo/rust',
     'examples/todo-mvc/rust',
     'examples/virtual_keyboard/rust',
     'examples/carousel/rust',
-    'examples/energy-monitor',
+    'demos/energy-monitor',
     'examples/mcu-board-support',
     'examples/uefi-demo',
     'examples/weather-demo',
```
Comment (ogoffart) [inline Cargo.toml:34]: weather-demo, as the same suggest, is another candidate to be moved.

(While uefi-demo is really an example, and not a demo.)

Comment (tronical) [reply Cargo.toml:34]: Yes. Preferably not in this PR as it's already getting big :)

**Comment by tronical** [comment]
CI and nightly test build are green.

**Review by ogoffart — APPROVED** [review]

File: examples/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ These examples demonstrate the main features of Slint and how to use them in dif
 
 
 
-| Thumbnail  | Description | Demo | 
+| Thumbnail  | Description | Example | 
 | --- | --- | --- | 
 | [Printer UI ![Printer Demo image](https://github.com/user-attachments/assets/7e7400ad-283a-4404-b04a-8620ba4df452)](./printerdemo) |  A fictional user interface for the touch screen of a printer. <br/> [Project...](./printerdemo) | [Wasm Demo](https://slint.dev/snapshots/master/demos/printerdemo/) |
```
Comment (ogoffart) [inline examples/README.md:11]: Need to remove this

Comment (tronical) [reply examples/README.md:11]: Oops, indeed. Thanks.

**Pull request merged**